inlanecruising_plugin/src/object_avoidance.cpp

```diff
@@ -0,0 +1,189 @@
+
+#include <inlanecruising_plugin/object_avoidance.h>
+
+namespace inlanecruising_plugin
+{
+namespace object_avoidance
+{
+    ObjectAvoidance::ObjectAvoidance(){}
+
+    cav_msgs::TrajectoryPlan ObjectAvoidance::update_traj_for_object(cav_msgs::TrajectoryPlan& original_tp, const carma_wm::WorldModelConstPtr& wm_, double current_speed_) 
+    {
+        
+        cav_msgs::TrajectoryPlan update_tpp_vector;
+
+        geometry_msgs::Twist current_velocity;
+        current_velocity.linear.x = current_speed_;
+
+        // // TODO get roadway object
+        std::vector<cav_msgs::RoadwayObstacle> rwol = wm_->getRoadwayObjects();
+        cav_msgs::RoadwayObstacleList rwol2;
+        rwol2.roadway_obstacles = rwol;
+
+        std::cout << "host_vehicle_size" << host_vehicle_size.x << std::endl;
+
+        std::vector<cav_msgs::RoadwayObstacle> rwol_collision = carma_wm::collision_detection::WorldCollisionDetection(rwol2, original_tp, host_vehicle_size, current_velocity, 10.0);// 10 replace to target_time
+        std::cout << "rwol_collision" << rwol_collision.size() << std::endl;
+        std::cout << "rwol" << rwol.size() << std::endl;
+
+        // correct the input types
+        if(rwol_collision.size() > 0) {
+
+            std::cout << "Collision!" << std::endl;
+
+            // use trajectory utiles to update trajectory plan points
+            
+            // lead vehicle trjactory
+            double dist_x = rwol_collision[0].object.pose.pose.position.x - original_tp.trajectory_points[0].x;
+            double dist_y = rwol_collision[0].object.pose.pose.position.y - original_tp.trajectory_points[0].y;
+            double x_lead = sqrt(dist_x*dist_x + dist_y*dist_y);
+
+            // roadway object position
+            double gap_time = (x_lead - x_gap)/current_speed_;
+
+            double collision_time = 0; //\TODO comming from carma_wm collision detection in future (not used now)
```

Comment:
If we need to update this in the future, let's get a Github and JIRA issue written up to follow up on these TODOs.

Reply:
created CAR 4288